--- mkinitrd-6.0.9/grubby/new-kernel-pkg	2007-12-15 17:41:08.258802271 +0200
+++ mkinitrd-6.0.24/grubby/new-kernel-pkg	2009-03-10 01:01:43.373402943 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Invoked upon installation or removal of a kernel package, the following
 # tasks are/can be done here:
@@ -10,6 +10,7 @@
 # 
 
 PATH=/sbin:/bin:$PATH
+PROGRAM=${0##*/}
 
 lilo=/sbin/lilo
 
@@ -114,15 +115,12 @@
 	return
     fi
 
-    # get the root filesystem to use; if it's on a label make sure it's
-    # been configured. if not, get the root device from mount
-    rootdevice=$(awk '{ if ($1 !~ /^[ \t]*#/ && $2 == "/") { print $1; }}' /etc/fstab)
-    short=$(echo $rootdevice | cut -d= -f1)
-    if [ "$short" == "LABEL" -o "$short" == "UUID" ]; then
-	device=$(echo resolveDevice "$rootdevice" | /sbin/nash  --forcequiet)
-	if [ -z "$device" ]; then
-	    rootdevice=$(mount | awk '$3 == "/" { print $1 }')
-	fi
+    # get the root filesystem to use
+    . /lib/geninitrd/functions
+    find_root /etc/fstab || exit
+    rootdevice=$rootdev
+    if [ -z "$rootdevice" ]; then
+	rootdevice=$(mount | awk '$3 == "/" { print $1 }')
     fi
 
     if [ -n "$mbkernel" -a -n "$cfgLilo" -a "$liloFlag" != "elilo" ]; then
@@ -208,10 +206,10 @@
 
 mkinitrd() {
     [ -n "$verbose" ] && echo "creating initrd $initrdfile using $version"
-    /sbin/mkinitrd --allow-missing -f $initrdfile $version
+    /sbin/geninitrd -f --initrdfs=rom $initrdfile $version
     rc=$?
     if [ $rc != 0 ]; then
-	echo "mkinitrd failed" >&2
+	echo "geninitrd failed" >&2
 	exit 1
     fi
 }
@@ -231,7 +229,6 @@
     [ -d /lib/modules/$version ] && rm -f /lib/modules/$version/modules.*    
 }
 
-
 while [ $# -gt 0 ]; do
     case $1 in
 	--mkinitrd)
