--- mkinitrd-6.0.86/./grubby/Makefile	2009-05-27 23:49:06.560272311 +0300
+++ mkinitrd-6.0.86/./grubby/Makefile	2009-05-27 23:52:01.606922232 +0300
@@ -31,9 +31,7 @@
 	-Wl,--wrap,open,--wrap,fopen,--wrap,opendir,--wrap,socket \
 	-Wl,--wrap,pipe
 
-grubby_LIBS = -lnash -lbdevid
-grubby_LIBS += -lparted -lblkid -luuid -lpopt -ldevmapper -lselinux -lsepol
-grubby_LIBS += $(shell pkg-config --libs libdhcp glib-2.0)
+grubby_LIBS = -lnash
 
 test: all
 	@export TOPDIR=$(TOPDIR)
--- mkinitrd-6.0.86/nash/Makefile~	2009-05-28 00:09:39.000000000 +0300
+++ mkinitrd-6.0.86/nash/Makefile	2009-05-28 00:10:56.307732260 +0300
@@ -37,15 +37,12 @@
 LDFLAGS += -Wl,-rpath-link,$(TOPDIR)/bdevid:$(TOPDIR)/nash
 
 nash_LIBPATHS = -L$(TOPDIR)/nash -L$(TOPDIR)/bdevid -L/$(LIB)
-nash_LIBS += -ldevmapper -lparted -lblkid -lselinux -lsepol
-nash_LIBS += -luuid -lpopt -lresolv -ldl -lelf
-nash_LIBS += $(shell pkg-config --libs libnl-1) -lm
-# We need to link against libgcc_s directly, or it'll /dlopen() it during
-# backtrace()!  This is not teh way!!!1!!!one!!!.
-nash_LIBS += -lgcc_s
+nash_LIBS += -ldevmapper -lparted -lblkid -ldl -lbdevid
+nash_prog_LIBS = -lnash -lresolv -lelf -lpopt
+nash_prog_LIBS += $(shell pkg-config --libs libnl-1)
 
 nash :: version.h $(OBJECTS) libnash.so
-	$(CC) $(LDFLAGS) -o $@ $(OBJECTS) $(nash_LIBPATHS) -lnash -lbdevid $(nash_LIBS)
+	$(CC) $(LDFLAGS) -o $@ $(OBJECTS) $(nash_LIBPATHS) $(nash_prog_LIBS)
 
 $(libnash_OBJECTS) :: CFLAGS += -fPIC
 
--- mkinitrd-6.0.86/bdevid/Makefile~	2009-05-28 00:26:05.000000000 +0300
+++ mkinitrd-6.0.86/bdevid/Makefile	2009-05-28 00:26:07.290963805 +0300
@@ -22,12 +22,10 @@
 
 bdevid_OBJECTS = bdevid.o module.o probe.o sysfs.o probelib.o
 bdevid_LIBS := -L. -L$(TOPDIR)/nash -ldl
-bdevid_LIBS += -lnash -lparted -lblkid -luuid -ldevmapper -lselinux -lsepol
 
 bdevidprobe_OBJECTS = probelib.o
 
 command_LIBS := -L. -L$(TOPDIR)/nash -lbdevid -ldl
-command_LIBS += -lnash -lparted -lblkid -luuid -ldevmapper -lselinux -lsepol
 
 MODULES = scsi ata usb
 OBJECTS = $(bdevid_OBJECTS)
@@ -45,15 +43,17 @@
 LDFLAGS += -Wl,-rpath-link,$(TOPDIR)/nash:$(TOPDIR)/bdevid
 LDFLAGS += -L$(TOPDIR)/nash -L$(TOPDIR)/bdevid
 
-bdevid :: command.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(command_LIBS)
+bdevid :: command.o ../nash/wrap.o libbdevid.so
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< ../nash/wrap.o $(command_LIBS)
 
 $(bdevid_OBJECTS) $(foreach o, $(MODULES), $(o).o) :: CFLAGS += -fPIC
 
 $(foreach lib,$(MODULES),$(lib).so) :: %.so : %.o ../nash/wrap.o libbdevidprobe.a
 	$(CC) $(LDFLAGS) -shared -Wl,-soname="$*.so.$(VERSION)" -o $@ $< ../nash/wrap.o $($*_LIBS) -lbdevidprobe
 
+libbdevid.so: libbdevid.so.$(VERSION)
+
 libbdevid.so.$(VERSION) : $(bdevid_OBJECTS)
 	$(CC) $(LDFLAGS) -shared -Wl,-soname="$(shell basename $@)" -o $@ $(bdevid_OBJECTS) $(bdevid_LIBS)
 
 libbdevidprobe.a : $(bdevidprobe_OBJECTS)
--- mkinitrd-6.0.86/bdevid/python/Makefile~	2009-05-28 00:27:36.000000000 +0300
+++ mkinitrd-6.0.86/bdevid/python/Makefile	2009-05-28 00:27:55.486773297 +0300
@@ -34,7 +34,7 @@
 LDFLAGS += -lpython$(PYVER) -ldl
 
 pybdevid_OBJECTS = pybdevid.o
-pybdevid_LIBS = -lbdevid -lnash -ldevmapper -lparted -lblkid
+pybdevid_LIBS = -lbdevid
 
 pybdevid.so :: $(pybdevid_OBJECTS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ -lbdevid $(pybdevid_LIBS)
