--- grubby-7.0.16/new-kernel-pkg~	2011-02-14 13:51:39.000000000 +0200
+++ grubby-7.0.16/new-kernel-pkg	2011-02-14 13:53:52.900519027 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Invoked upon installation or removal of a kernel package, the following
 # tasks are/can be done here:
@@ -139,7 +139,9 @@
     fi
 
     # get the root filesystem to use
-    rootdevice=$(awk '{ if ($1 !~ /^[ \t]*#/ && $2 == "/") { print $1; }}' /etc/fstab)
+    . /lib/geninitrd/functions
+    find_root /etc/fstab || exit
+    rootdevice=$rootdev
 
     if [ -n "$mbkernel" -a -n "$cfgLilo" -a "$liloFlag" != "elilo" ]; then
 	[ -n "$verbose" ] && echo "multiboot specified, not updating lilo.conf"
@@ -289,6 +291,8 @@
 mkinitrd() {
     if [ -n "$dracut" ]; then
         tool="/sbin/dracut $dracuthostonly -f $initrdfile $version"
+    elif [ -x /sbin/geninitrd ]; then
+	tool="/sbin/geninitrd -f $initrdfile $version"
     else
 	tool="/sbin/mkinitrd --allow-missing -f $initrdfile $version"
     fi
