--- grubby-8.3/grubby.c~	2011-09-29 23:38:12.000000000 +0300
+++ grubby-8.3/grubby.c	2011-10-26 12:10:12.124419102 +0300
@@ -157,7 +157,7 @@
 };
 
 struct configFileInfo grubConfigType = {
-    .defaultConfig = "/etc/grub.conf",
+    .defaultConfig = "/boot/grub/menu.lst",
     .keywords = grubKeywords,
     .defaultIsIndex = 1,
     .defaultSupportSaved = 1,
--- grubby-8.3/new-kernel-pkg~	2011-09-29 23:38:12.000000000 +0300
+++ grubby-8.3/new-kernel-pkg	2011-10-26 12:11:06.743604439 +0300
@@ -91,7 +91,7 @@
 else
     # this leaves i?86 and x86_64
     liloConfig=/etc/lilo.conf
-    grubConfig=$(readlink -f /etc/grub.conf 2>/dev/null)
+    grubConfig=/boot/grub/menu.lst
     grub2Config=$(readlink -f /etc/grub2.cfg 2>/dev/null)
     grub2EfiConfig=$(readlink -f /etc/grub2-efi.cfg 2>/dev/null)
     bootPrefix=/boot
