--- mkinitrd-6.0.86/grubby/grubby.c~	2009-05-21 22:14:01.000000000 +0300
+++ mkinitrd-6.0.86/grubby/grubby.c	2009-05-27 23:36:24.400378963 +0300
@@ -140,7 +140,7 @@
 };
 
 struct configFileInfo grubConfigType = {
-    "/etc/grub.conf",			    /* defaultConfig */
+    "/boot/grub/menu.lst",			    /* defaultConfig */
     grubKeywords,			    /* keywords */
     1,					    /* defaultIsIndex */
     1,					    /* defaultSupportSaved */
--- mkinitrd-6.0.86/grubby/new-kernel-pkg~	2009-05-21 22:14:01.000000000 +0300
+++ mkinitrd-6.0.86/grubby/new-kernel-pkg	2009-05-27 23:36:55.383790363 +0300
@@ -73,7 +73,7 @@
 else
     # this leaves i?86 and x86_64
     liloConfig=/etc/lilo.conf
-    grubConfig=$(readlink -f /etc/grub.conf 2>/dev/null)
+    grubConfig=/boot/grub/menu.lst
     bootPrefix=/boot
     liloFlag=lilo
     isx86="yes"
